sdk/core/typespec_client_core/src/http/options/retry.rs

```diff
@@ -56,37 +60,73 @@ impl Default for RetryMode {
 pub struct RetryOptions {
     /// The algorithm to use for calculating retry delays.
     mode: RetryMode,
+
+    /// The status codes that should trigger retries. If `None`, the default set of status codes
+    /// will be used as described in the documentation for [`RetryOptions::with_retry_statuses`].
+    status_codes: Option<Vec<StatusCode>>,
 }
 
 impl RetryOptions {
     /// A retry strategy where attempts happen at intervals that get exponentially longer with each retry.
     pub fn exponential(options: ExponentialRetryOptions) -> Self {
         Self {
             mode: RetryMode::Exponential(options),
+            status_codes: None,
         }
     }
 
     /// A retry strategy where attempts happen at fixed intervals; each delay is a consistent duration.
     pub fn fixed(options: FixedRetryOptions) -> Self {
         Self {
             mode: RetryMode::Fixed(options),
+            status_codes: None,
         }
     }
 
     /// A custom retry using the supplied retry policy.
     pub fn custom<T: RetryPolicy + 'static>(policy: Arc<T>) -> Self {
         Self {
             mode: RetryMode::Custom(policy),
+            status_codes: None,
         }
     }
 
     /// No retries will be attempted.
     pub fn none() -> Self {
         Self {
             mode: RetryMode::None,
+            status_codes: None,
         }
     }
 
+    /// Set status codes that should trigger retries.
+    ///
+    /// By default, retries are attempted for these status codes:
+    /// - 408 (Request Timeout)
+    /// - 429 (Too Many Requests)
+    /// - 500 (Internal Server Error)
+    /// - 502 (Bad Gateway)
+    /// - 503 (Service Unavailable)
+    /// - 504 (Gateway Timeout)
+    ///
+    /// Use this method to override the default set. Passing an empty
+    /// vector will disable status-based retries.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use typespec_client_core::http::{RetryOptions, ExponentialRetryOptions, StatusCode};
+    /// let opts = RetryOptions::exponential(ExponentialRetryOptions::default())
+    ///     .with_retry_statuses(vec![
+    ///         StatusCode::ServiceUnavailable,
+    ///         StatusCode::TooManyRequests,
+    ///     ]);
+    /// ```
+    pub fn with_retry_statuses(mut self, statuses: Vec<StatusCode>) -> Self {
+        self.status_codes = Some(statuses);
+        self
+    }
+
     pub(crate) fn to_policy(&self, retry_headers: RetryHeaders) -> Arc<dyn Policy> {
```

Comment:
I know it's an existing call, but should `to_policy` consume self rather than borrowing self? Possibly after renaming it `into_policy` to avoid a clippy warning?

The `.clone()` below just annoys me :).